### DISCUSSION TITLE
How can I disable compositing?

### DISCUSSION THREAD

**Discussion by BlueNebulaDev** [nix-community/plasma-manager#404] [Q&A] [answered]
KDE Plasma 6 offers an option to disable compositing. In System Settings > Compositor, "Enable on startup".

Is it possible to disable it via plasma-manager?

**Comment by HeitorAugustoLN** [comment]
I guess this is available on X11 only? I can't find it on my system. It should be possible to disable it if an option is provided. I will take a look into it.

**Reply by BlueNebulaDev** [reply]
I'm indeed using plasmax11.

I believe it can be disabled by adding this to `~/.config/kwinrc`:
```
[Compositing]
Enabled=false
```

See https://invent.kde.org/plasma/kwin/-/blob/master/src/kwin.kcfg#L239-274

**Reply by HeitorAugustoLN** [reply, marked as answer]
That's right, so you could just set like this in your configuration:
```nix
{
  programs.plasma.configFile.kwinrc.Compositing.Enabled = false;
}
```

**Reply by BlueNebulaDev** [reply]
Ohh, thanks. Couldn't find that option.

Would you know if it's possible to add custom settings to the hotkeys configuration as well?  
I'd like to add `StartupNotify=false` to the [Desktop Entry] plasma-manager creates for `hotkeys.commands`...
Otherwise, should I open an issue, or a new discussion, or directly submit a PR, or what else, for something like this?

**Reply by HeitorAugustoLN** [reply]
You could add it directly in home-manager using:
```nix
{
  xdg.desktopEntries.plasma-manager-commands.startupNotify = false;
}
```

But if you want it to be set directly using plasma-manager, I can do it in a few days, or you can submit an PR.
